Shared/Objects/FilterDrawerButton.swift

```diff
@@ -0,0 +1,101 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2023 Jellyfin & Jellyfin Contributors
+//
+
+import Defaults
+import Foundation
+
+enum FilterDrawerButton: String, CaseIterable, Defaults.Serializable, Displayable, Identifiable {
+
+    case filters
+    case genres
+    case order
+    case sort
+
+    var displayTitle: String {
+        switch self {
+        case .filters:
+            return L10n.filters
+        case .genres:
+            return L10n.genres
+        case .order:
+            return L10n.order
+        case .sort:
+            return L10n.sort
+        }
+    }
+    
+    var id: String {
+        rawValue
+    }
+
+    var settingsFilter: WritableKeyPath<ItemFilters, [ItemFilters.Filter]> {
+        switch self {
+        case .filters:
+            return \.filters
+        case .genres:
+            return \.genres
+        case .order:
+            return \.sortOrder
+        case .sort:
+            return \.sortBy
+        }
+    }
+    
+    var settingsSelectorType: SelectorType {
+        switch self {
+        case .filters, .genres:
+            return .multi
+        case .order, .sort:
+            return .single
+        }
+    }
+    
+    var settingsItemsFilterProperty: String {
+        switch self {
+        case .filters:
+             return "filters"
+        case .genres:
+            return "genres"
+        case .order:
+            return "sortOrder"
+        case .sort:
+            return "sortBy"
+        }
+    }
+    
+    func isItemsFilterActive(viewModel: FilterViewModel) -> Bool {
```

Comment:
We should not reference a view model, we want to reference the `ItemFilters` object instead using the key path defined above. I actually liked defining the "inactive/default" state that's defined and then checking that. Additionally, that state could be used to initialize the `ItemFilters` properties.

```swift
var genres: [Filter] = FilterDrawerButton.genres.default
var filters: [Filter] = // etc
var sortOrder: [Filter] = // etc
var sortBy: [Filter] = // etc
```

Reply:
I've re-added the defaults and am instead passing in the itemFilters. I am a little confused by your declaration idea. In the enum, they are a  case so I was getting an error when trying to convert them to var. That being said, I think I'm just looking in the wrong place. Below is what I currently have. It feels a little repetitive  but I think changing the setting in the enum oppposed to changing the function:

```
    var itemFilterDefault: [ItemFilters.Filter] {
        switch self {
        case .filters:
            return []
        case .genres:
            return []
        case .order:
            return [APISortOrder.ascending.filter]
        case .sort:
            return [SortBy.name.filter]
        }
    }
    
    func isItemsFilterActive(activeFilters: ItemFilters) -> Bool {
        switch self {
        case .filters:
            return activeFilters.filters != self.itemFilterDefault
        case .genres:
            return activeFilters.genres != self.itemFilterDefault
        case .order:
            return activeFilters.sortOrder != self.itemFilterDefault
        case .sort:
            return activeFilters.sortBy != self.itemFilterDefault
        }
    }
```
Let me know if there is a better way for me to be doing this. 100% this is just me putting this in the wrong place.